src/main/scala/cognite/spark/v1/RawTableRelation.scala

```diff
@@ -287,22 +288,13 @@ class RawTableRelation(
     dfWithUnRenamedKeyColumns.foreachPartition((rows: Iterator[Row]) => {
       config.maxOutstandingRawInsertRequests match {
         case Some(maxOutstandingRawInsertRequests) =>
-          // We first group by batch size of a write, and then group that by the number of allowed parallel
-          // outstanding requests to avoid queueing up too many requests towards the RAW API (and this potentially
-          // leading to an OutOfMemory)
-          // Note: This is a suboptimal fix, as if one of the requests in a batch is slow, we will not
-          // start on the next batch (this limitation used to be per partition). Instead, we should
-          // have a cats.effect.std.Semaphore permit with X number of outstanding requests
-          // or cats.effect.concurrent.Backpressure.
-          rows
-            .grouped(batchSize)
-            .toSeq
-            .grouped(maxOutstandingRawInsertRequests)
-            .foreach { batch =>
-              batch.toVector
-                .parTraverse_(postRows(columnNames, _))
-                .unsafeRunSync()
+          Backpressure[IO](Backpressure.Strategy.Lossless, maxOutstandingRawInsertRequests)
+            .flatMap { backpressure =>
+              rows.grouped(batchSize).toVector.parTraverse_ { batch: Seq[Row] =>
```

Comment:
compared to `_.grouped.toSeq.grouped.foreach` it will now create a vector of all batches, they will reference all Row items from iterator which potentially could be not a lot of mem, less than request bodies, more than constant mem

but compared to `.grouped.toVector` it is about the same apart from having semaphore